src/ApiCore/Transport/GrpcTransport.php

```diff
@@ -174,16 +174,20 @@ public function startUnaryCall(Call $call, array $options)
         );
 
         $promise = new Promise(
-            function () use ($call, &$promise) {
-                list($response, $status) = $call->wait();
+            function () use ($unaryCall, $options, &$promise) {
+                list($response, $status) = $unaryCall->wait();
 
                 if ($status->code == Code::OK) {
+                    if (isset($options['metadataCallback'])) {
```

Comment:
Is this something we would want any transport implementation to have support for? If so, do you think it would be worth adding something that handles this logic to the contract?